refactoring: replace pool conditionals with registry-based quicklinks helper

zcht opened the pull request:
This PR refactors the Home page by extracting pool-specific logic from home.component.ts into a dedicated helper module (home.quicklinks.ts).

A previously long if / else if chain inside the component has been replaced with a declarative pool registry, improving readability, maintainability, and extensibility.

At the same time, pool icon handling was redesigned to be robust, privacy-friendly, and fully centralized.

---

* Component API cleanup
* Improved Home card header layout
* Design rationale

## Extract pool logic from HomeComponent

* Removed pool-specific if / else if logic from home.component.ts
* Introduced home.quicklinks.ts as a single source of truth for:
** pool detection
** dashboard / statistics link generation
** capability flags (e.g. ping support)
** icon / favicon resolution

## Replace else if chain with registry-based rules

* Pool detection is now done via a first-match-wins registry
* Each pool entry declares:
** how it is matched
** how quick links are built
** optional overrides (capabilities, icons)
* Unknown pools are handled gracefully via sensible defaults

## Centralized and robust pool icon handling

Pool icons are resolved using a strict priority order:

1. Manual icon override (iconUrl)
2. Custom favicon path with optional hostname override (faviconPath, faviconHost)
3. Automatic favicon fallback (<host>/favicon.ico)
4. Local default icon for non-web/local pools

## Local pool handling via registry rule

* Local endpoints (localhost, private IPs, .local, IPv6 ::1) are detected via a dedicated registry entry
* These pools always resolve to a local default icon
* Prevents invalid HTTPS favicon requests against non-web services



---

<img width="1154" height="256" alt="image" src="https://github.com/user-attachments/assets/06a3ebad-d422-455d-859b-6660f0a0ba4b" />
<img width="1150" height="254" alt="image" src="https://github.com/user-attachments/assets/25402356-8396-4bfa-8ca8-ee020c39e132" />

zcht committed:
add centralized pool quicklink helper

### What
Added `home.quicklinks.ts`, a centralized helper for pool-related logic.

### Why
Pool-specific URL matching and capabilities do not belong in the component.
Extracting them:
- improves maintainability
- avoids fragile string matching on raw stratum URLs
- makes future pool additions data-driven and testable

zcht committed:
refactor: delegate pool quicklink logic to helper

### What
Refactored `home.component.ts` to use the new quicklinks helper.

### Why
The HomeComponent should focus on UI state and streams.
Delegating pool-specific logic:
- reduces complexity in the component
- avoids duplication and conditional chains
- keeps behavior consistent across primary and fallback pools

zcht committed:
add robust pool icon resolution with local pool fallback

* Add registry-based local pool detection (localhost, private IPs, .local)
* Prevent broken favicon requests for local stratum endpoints
* Support manual icon overrides per pool
* Support custom favicon paths (svg/png) and hostname overrides
* Fall back to a local default icon for non-web pool endpoints

This keeps icon logic out of components and avoids template error handling.

zcht committed:
move pool icon before header title

* Render pool icon before the header title
* Use flex layout for consistent alignment
* Remove space-between layout to avoid unintended positioning
* Ensure long translations and dual-pool labels behave correctly

zcht committed:
add poolIconUrl helper for header icon rendering

Expose a small wrapper method on HomeComponent to resolve pool icons for template bindings.

* Add `poolIconUrl(host)` as a thin delegate to home.quicklinks.ts
* Keep all icon resolution logic centralized and reusable
* Avoid direct helper imports inside templates
* Prepare component for clean icon rendering in the card header

No behavior change besides enabling template access to icon URLs.

zcht committed:
remove d-central pool reference

... because the pool no longer exists.

zcht committed:
copy fix

shufps commented:
looks good :+1: 

Default icon must be made neutral and not plebbase style though :wink:

zcht committed:
default icon replaced

zcht committed:
PowerMining pool added for recognition

zcht commented:
Oh crap... forgot to import the stash. Wanted to test a bi-color icon during testing. It was supposed to be a BTC icon, but now that I think about it, it doesn't really make much sense because other sha256 coins can also be mined.

I've now chosen this icon, which should be neutral enough for everyone... but in BTC colors :D

+ added powerming pool

<img width="1302" height="205" alt="image" src="https://github.com/user-attachments/assets/ae7dc350-99d4-4af9-9d23-dc0edea44ced" />

zcht committed:
somewhat tidied up

zcht committed:
custom CK Pool icons added depending on region

zcht committed:
icon reference for ck pool

zcht committed:
missing recognition for CK pool Australia added

zcht committed:
recognition of a pool on Umbrel via local DNS...

... if the Umbrel instance is accessible via the local DNS entry and the pool is assigned via this entry, the Umbrel icon for the pool is displayed.

Otherwise, the local pool icon is displayed, e.g., if the pool is accessed via a local IP address.

zcht committed:
Increased icon output size to 21px

zcht committed:
default icon replaced with more appropriate schematic visual output

zcht committed:
dedicated public pool icon inserted...

... since public pool blocks ICMP, at least in my network, the icon cannot be extracted. For this reason, a locally stored icon is simply output to avoid the ICMP problem.

zcht committed:
manually set faviconPath for Atlaspool

zcht commented:
I think everything should be fine now. I couldn't access public pool, it just doesn't work for me, so I added a dedicated favicon.
 
I added an implementation specifically for Umbrel. I changed the default icon, which should be more semantically appropriate.

For CK pool, I also added icons manually because there is nothing on the website and different instances are offered depending on the region.

Otherwise, I tested all stored pools again in detail.

@shufps I think that should be fine now.

<img width="1301" height="1253" alt="image" src="https://github.com/user-attachments/assets/60db9548-016f-4e03-afea-d6dcd2081831" />

zcht committed:
ensure default pool icon fallback for unknown pools

... adds an image error fallback to ensure a default pool icon is shown for unknown or unreachable pool favicons. Registry and quicklinks remain unchanged.

zcht committed:
use distinct default icon for unknown public pools

... local pools continue to use the existing standard pool icon, while public pools that are not present in the registry fall back to a special “public pool” icon if no favicon can be loaded.

zcht commented:
With unit tests, some things probably wouldn't have happened... well, now it's 99.9% functional... :D

I also implemented an icon for public external pools that are not in the Nerd*OS pool registry and are NOT local pools. This way, you can quickly see where you are by looking at the icon.

So, at the moment, we have the following icons:
* Umbrel via local DNS -> umbrella icon
* Local pool -> local pool icon
* Known pool from the Nerd*OS registry -> stored icon or favicon or default icon
* Unknown public pool -> globe icon

All icons are also directly linked to the pool, and for known pools from the registry, directly to the dashboard of the respective BTC address.

<img width="1193" height="424" alt="image" src="https://github.com/user-attachments/assets/bc38807e-3ab5-41ee-b89d-d55c299d6f15" />
